src/praisonai-bot/praisonai_bot/bots/_pairing_ui.py

```
@@ -18,11 +18,36 @@
+def _persisted_callback_secret() -> str:
+    """Return the persisted gateway signing secret (cached read).
+
+    Resolves through the same auto-provisioned, owner-only secret store used
+    by the pairing code path so approve/deny buttons survive process restarts.
+    Falls back to a per-process random secret only if the store is unwritable.
+    """
+    try:
+        from praisonai_bot.gateway.pairing import _DEFAULT_STORE_DIR, _load_or_create_secret
+
+        return _load_or_create_secret(_DEFAULT_STORE_DIR).decode()
+    except Exception:  # pragma: no cover - defensive fallback
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)